x-django-service: &django-service
  user: "${UID}:${GID}"
  build: .
  volumes:
    - .:/code
  depends_on:
    db:
      condition: service_healthy
  environment:
    PYTHONUNBUFFERED: 1
    DATABASE_URL: "postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@db/${POSTGRES_DB}"
    DJANGO_SECRET_KEY: "${DJANGO_SECRET_KEY}"

services:
  db:
    image: postgres:14
    environment:
      POSTGRES_USER: "${POSTGRES_USER}"
      POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
      POSTGRES_DB: "${POSTGRES_DB}"
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}" ]
      interval: 5s
      timeout: 5s
      retries: 5

  web:
    <<: *django-service
    command: python manage.py runserver 0.0.0.0:8000
    extra_hosts:
      - "host.docker.internal:host-gateway"
    ports:
      - "8000:8000"

  worker:
    <<: *django-service
    command: python manage.py worker
    extra_hosts:
      - "host.docker.internal:host-gateway"

volumes:
  postgres_data:
    driver: local
